src/state/registers.ts

```diff
@@ -583,6 +585,33 @@ export abstract class RegisterSet implements vscode.Disposable {
 
     this._named.set("", this.null);
     this._named.set("null", this.null);
+
+    // Watch systemClipboardRegister setting and update binding
+    if (_extension) {
+      _extension.observePreference<keyof Registers>(
+        ".systemClipboardRegister",
+        (value) => {
+          // Reset old value to be a GeneralRegister
+          if (this._systemClipbordRegister === "dquote") {
+            this.dquote = new GeneralPurposeRegister('"', "copy");
+            this._named.set("dquote", this.dquote);
+          } else if (this._systemClipbordRegister != null) {
+            this._named.set(this._systemClipbordRegister, new GeneralPurposeRegister(this._systemClipbordRegister, "copy"));
+          }
+
+          // Set new value to be a ClipboardRegister
+          if (value === "dquote") {
+            this.dquote = new ClipboardRegister();
+            this._named.set("dquote", this.dquote);
+          } else if (value != null) {
+            this._named.set(value, new ClipboardRegister());
```

Comment:
This will lead to strange behaviors if `value` corresponds to a built-in register that is given a property, e.g. ":". Please validate that `value` is a single letter.

Reply:
I've moved the available options for the clipboard register into `meta.ts` so we can check that here to ensure that only valid options are processed.